File: JavaScript/leetcode_1436.js
/**
 * @param {string[][]} paths
 * @return {string}
 */
var destCity = function(paths) {
    let pset = new Set();
    for (let index = 0; index < paths.length; index++) {
        pset.add(paths[index][0])
    }
    for (let index = 0; index < paths.length; index++) {
        if (!pset.has(paths[index][1])) return paths[index][1]
    }
};

File: Python/leetcode_1436.py
from typing import List


def dest_city(paths: List[List[str]]) -> str:
    pset: set[str] = set()
    for d in paths:
        pset.add(d[0])
    for d in paths:
        if d[1] not in pset:
            return d[1]
    return paths[-1][-1]
